mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -197,8 +238,86 @@ class LocalLDAModel private[clustering] (
   // TODO:
   // override def topicDistributions(documents: RDD[(Long, Vector)]): RDD[(Long, Vector)] = ???
 
+  /**
+   * Calculate and return per-word likelihood bound, using the `batch` of
+   * documents as evaluation corpus.
+   */
+  // TODO: calcualte logPerplexity over training set online during training, reusing gammad instead
+  // of performing variational inference again in [[bound()]]
+  def logPerplexity(
+      batch: RDD[(Long, Vector)],
+      totalDocs: Long): Double = {
```

Comment:
Don't think I understand what the distinction between "average" and "total" perplexity is.

I removed this argument since we can just do `documents.count()` inside the method.